vpr/src/base/SetupGrid.cpp

```diff
@@ -150,6 +150,17 @@ static DeviceGrid auto_size_device_grid(const std::vector<t_grid_def>& grid_layo
 
         VTR_ASSERT_SAFE_MSG(std::find_if(auto_layout_itr + 1, grid_layouts.end(), is_auto_grid_def) == grid_layouts.end(), "Only one <auto_layout>");
 
+        //Determine maximum device size to try before concluding that the circuit cannot fit on any device
+        //Calculate total number of required instances
+        //Then multiply by a factor of 100 as overhead
```

Comment:
I think you're multiplying by a factor of 1000 (should update comment).
Should say why you're doing this -- to avoid infinite loops if increasing the grid size never gets you more of the instance type you need and hence never lets you fit.